### DISCUSSION TITLE
Can you run RepoMix on a C# repo?

### DISCUSSION THREAD

**Discussion by ratacat** [yamadashy/repomix#204] [Q&A]
I mean, clearly it's written in Node. I'm just wondering if it supports other things?

**Comment by yamadashy** [comment]
Hi, @ratacat !
Yes! Repomix can process any repository regardless of the programming language.

The only requirement is having Node.js v16+ installed on your local machine. Once you have Node.js, you can run `npx repomix` on any folder or repository.

**Comment by Adit-Jain-srm** [comment]
Yes, Repomix works on **any programming language** — C#, Java, Python, Rust, Go, Ruby, PHP, whatever. It's language-agnostic.

Repomix doesn't compile or analyze your code semantically — it reads files, applies formatting (XML/Markdown/plain text output), respects `.gitignore` and custom ignore patterns, and packages everything into a single file for LLM consumption.

For C# repos specifically:

```bash
# From your C# project root
repomix

# Or target a specific directory
repomix --include "src/**/*.cs"

# Exclude test/build artifacts
repomix --ignore "bin/**,obj/**,*.Designer.cs"
```

The output will contain all your `.cs`, `.csproj`, `.sln` files (unless excluded) formatted for pasting into an LLM context window.
